File: app/src/test/java/org/schulcloud/mobile/data/model/responseBodies/SignedUrlResponseTest.java
package org.schulcloud.mobile.data.model.responseBodies;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class SignedUrlResponseTest {

    private static final String URL = "URL";
    private static final String CONTENT_TYPE = "CONTENT_TYPE";
    private static final String NAME = "NAME";
    private static final String PATH = "PATH";
    private static final String THUMBNAIL = "THUMBNAIL";

    private SignedUrlResponse signedUrlResponse;

    @Before
    public void setUp() throws Exception {
        signedUrlResponse = newSignedUrlResponse();
    }

    @Test
    public void testForConstructor() {
        assertEquals(signedUrlResponse.url, URL);
        assertEquals(signedUrlResponse.header.getContentType(), CONTENT_TYPE);
        assertEquals(signedUrlResponse.header.getMetaName(), NAME);
        assertEquals(signedUrlResponse.header.getMetaPath(), PATH);
        assertEquals(signedUrlResponse.header.getMetaThumbnail(), THUMBNAIL);
    }

    public static SignedUrlResponse newSignedUrlResponse() {
        SignedUrlResponse signedUrlResponse = new SignedUrlResponse();
        signedUrlResponse.url = URL;
        SignedUrlResponse.SignedUrlResponseHeader header = new SignedUrlResponse.SignedUrlResponseHeader();
        header.setContentType(CONTENT_TYPE);
        header.setMetaName(NAME);
        header.setMetaPath(PATH);
        header.setMetaThumbnail(THUMBNAIL);

        signedUrlResponse.header = header;
        return signedUrlResponse;
    }

}

File: app/src/test/java/org/schulcloud/mobile/models/file/SignedUrlResponseTest.kt
package org.schulcloud.mobile.models.file

import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class SignedUrlResponseTest {
    private companion object {
        const val URL = "url"
        const val CONTENTTYPE = "contentType"
        const val METAPATH = "metaPath"
        const val METANAME = "metaName"
        const val METAFLATNAME = "metaFlatName"
        const val METATHUMBNAIL = "metaThumbnail"
    }

    private val newSignedUrlResponse: SignedUrlResponse
        get() = SignedUrlResponse().apply {
            url = URL
            header = SignedUrlResponse.SignedUrlResponseHeader().apply {
                contentType = CONTENTTYPE
                metaPath = METAPATH
                metaName = METANAME
                metaFlatName = METAFLATNAME
                metaThumbnail = METATHUMBNAIL
            }
        }
    private lateinit var signedUrlResponse: SignedUrlResponse

    @Before
    fun setUp() {
        signedUrlResponse = newSignedUrlResponse
    }

    @Test
    fun testGetProperties() {
        assertEquals(URL, signedUrlResponse.url)
        assertEquals(CONTENTTYPE, signedUrlResponse.header?.contentType)
        assertEquals(METAPATH, signedUrlResponse.header?.metaPath)
        assertEquals(METANAME, signedUrlResponse.header?.metaName)
        assertEquals(METAFLATNAME, signedUrlResponse.header?.metaFlatName)
        assertEquals(METATHUMBNAIL, signedUrlResponse.header?.metaThumbnail)
    }
}
